collector/collector.go

```diff
@@ -466,11 +478,33 @@ PduLoop:
 		}
 	}
 	ch <- prometheus.MustNewConstMetric(
-		prometheus.NewDesc("snmp_scrape_duration_seconds", "Total SNMP time scrape took (walk and processing).", nil, nil),
+		prometheus.NewDesc("snmp_scrape_duration_seconds", "Total SNMP time scrape took (walk and processing).", nil, moduleLabel),
 		prometheus.GaugeValue,
 		time.Since(start).Seconds())
 }
 
+// Collect implements Prometheus.Collector.
+func (c Collector) Collect(ch chan<- prometheus.Metric) {
+	var wg sync.WaitGroup
+	sem := make(chan struct{}, c.concurrency)
+	for name, module := range c.modules {
+		sem <- struct{}{}
+		wg.Add(1)
```

Comment:
I'm not sure this concurrency approach is how I would do it.

The more common pattern is to startup a pool of worker routines and then send the module list down the channel to them.

Something [like this in node_exporter](https://github.com/prometheus/node_exporter/blob/37ce0bab8c7afe5133b33bb44cff06fe965cf109/collector/filesystem_linux.go#L58-L73).

What do you think?

Reply:
I thought it was a simple implementation for me, but I can change it if you want to match other implementations.

Reply:
Yes, I think the other method is easier to read/understand.